service/internal/integrationtest/s3/presign_test.go

```diff
@@ -0,0 +1,156 @@
+// +build integration
+
+package s3
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/service/internal/integrationtest"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestInteg_PresignURL(t *testing.T) {
+	cases := map[string]struct {
+		body                 io.Reader
+		expires              time.Duration
+		sha256Header         string
+		expectedSignedHeader http.Header
+	}{
+		"standard": {
+			body: bytes.NewReader([]byte("Hello-world")),
+			expectedSignedHeader: http.Header{
+				"content-type":   {"application/octet-stream"},
```

Comment:
This header probably shouldn't be set by default for put objects.  This header being set requires users of the Presign URL to use the signed header in addition to the URL.

Reply:
We should have a release candidate issue to track if the `content-type` header auto population should be removed being set for streams by default. Having this header hear make the presign URL more difficult to use because the map of headers have to be passed to the downstream client using the presigned URL.

Reply:
Created https://github.com/aws/aws-sdk-go-v2/issues/923 to track this